test/e2e/clusterctl_upgrade_test.go

```diff
@@ -62,24 +62,24 @@ var _ = Describe("When testing clusterctl upgrades (v1.4=>current)", func() {
 			ArtifactFolder:         artifactFolder,
 			SkipCleanup:            skipCleanup,
 			InfrastructureProvider: pointer.String("docker"),
-			InitWithBinary:         "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.4.5/clusterctl-{OS}-{ARCH}",
+			InitWithBinary:         "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.5.4/clusterctl-{OS}-{ARCH}",
 			// We have to pin the providers because with `InitWithProvidersContract` the test would
-			// use the latest version for the contract (which is v1.5.X for v1beta1).
-			InitWithCoreProvider:            "cluster-api:v1.4.5",
-			InitWithBootstrapProviders:      []string{"kubeadm:v1.4.5"},
-			InitWithControlPlaneProviders:   []string{"kubeadm:v1.4.5"},
-			InitWithInfrastructureProviders: []string{"docker:v1.4.5"},
+			// use the latest version for the contract (which is v1.6.X for v1beta1).
+			InitWithCoreProvider:            "cluster-api:v1.5.4",
+			InitWithBootstrapProviders:      []string{"kubeadm:v1.5.4"},
+			InitWithControlPlaneProviders:   []string{"kubeadm:v1.5.4"},
+			InitWithInfrastructureProviders: []string{"docker:v1.5.4"},
 			InitWithProvidersContract:       "v1beta1",
-			// NOTE: If this version is changed here the image and SHA must also be updated in all DockerMachineTemplates in `test/e2e/data/infrastructure-docker/v1.4/bases.
-			InitWithKubernetesVersion: "v1.27.3",
+			// NOTE: If this version is changed here the image and SHA must also be updated in all DockerMachineTemplates in `test/e2e/data/infrastructure-docker/v1.5/bases.
+			InitWithKubernetesVersion: "v1.28.0",
 			WorkloadKubernetesVersion: "v1.27.3",
```

Comment:
We should also bump the WorkloadKubernetesVersion to 1.28.0

(same for the next test case)

Reply:
We should probably add this comment from the release tasks here inline as a godoc comment:
> Note: Both InitWithKubernetesVersion and WorkloadKubernetesVersion should be the highest mgmt cluster version supported by the respective Cluster API version.

Reply:
pushed up some changes, i believe they accomplish the suggestions (please let me know if otherwise)

Reply:
Looks perfect. Thank you!